metadata/elasticsearch/src/main/java/com/spotify/heroic/metadata/elasticsearch/MetadataBackendKV.java

```diff
@@ -437,36 +433,42 @@ public boolean isReady() {
     protected Series toSeries(SearchHit hit) {
         final Map<String, Object> source = hit.getSourceAsMap();
         final String key = (String) source.get(KEY);
+
+        @SuppressWarnings("unchecked")
         final Iterator<Map.Entry<String, String>> tags =
             ((List<String>) source.get(TAGS)).stream().map(this::buildTag).iterator();
+
         return Series.of(key, tags);
     }
 
     private <T, O> AsyncFuture<O> entries(
-        final Filter filter,
-        final OptionalLimit limit,
+        final FindSeries.Request seriesRequest,
         final Function<SearchHit, T> converter,
-        final Transform<ScrollTransformResult<T>, O> collector,
+        final Transform<SearchTransformResult<T>, O> collector,
         final Consumer<SearchRequest> modifier
     ) {
-        final QueryBuilder f = filter(filter);
+        final QueryBuilder f = filter(seriesRequest.getFilter());
+        OptionalLimit limit = seriesRequest.getLimit();
 
         return doto(c -> {
             SearchRequest request = c.getIndex().search(METADATA_TYPE);
-            request.scroll(SCROLL_TIME);
             request.source()
                 .size(limit.asMaxInteger(scrollSize))
-                .query(new BoolQueryBuilder().must(f));
+                .query(new BoolQueryBuilder().must(f))
+                .sort(new FieldSortBuilder("_doc"));
 
             modifier.accept(request);
 
-            AsyncFuture<ScrollTransformResult<T>> scroll = scrollEntries(
-                c, request, limit, converter);
-
-            return scroll
-                .onResolved(r -> ofNullable(r.getLastScrollId())
-                    .ifPresent(c::clearSearchScroll))
-                .directTransform(collector);
+            if (seriesRequest.getFeatures().hasFeature(Feature.METADATA_LIVE_CURSOR)) {
```

Comment:
shouldn't this be `!seriesRequest.getFeatures().hasFeature(Feature.METADATA_LIVE_CURSOR)` ? 

otherwise this will use scroll when feature METADATA_LIVE_CURSOR is enabled?

Reply:
Yep I have the logic backwards, will fix.